### ISSUE TITLE
Compiling on Catalina

### ISSUE THREAD

**Issue by gcstr** [lwouis/alt-tab-macos#162]
I'm trying to compile the project locally, but I'm stuck on Catalina's Screen Recording permission.
I do have the permission set on System Preferences but still, I get the error message

```
"Before using this app, you need to give permission in System Preferences > Security & Privacy > Privacy > Accessibility."
"Please authorize and re-launch."
"See https://help.rescuetime.com/article/59-how-do-i-enable-accessibility-permissions-on-mac-osx"
```

<img width="780" alt="Screenshot 2020-02-12 at 01 45 37" src="https://user-images.githubusercontent.com/62089/74292851-68814b80-4d39-11ea-8fef-c36536c80243.png">

Also, I set the Signing Certificate to `Development` on my personal team on my local project settings, but I still can't pass this stage.

Is there anything else I can do to set this permission and run the project locally?

PS: I also tried https://github.com/lwouis/alt-tab-macos/issues/52 ticking and unticking several times/builds, but nothing seems to help.

**Comment by lwouis** [comment]
Hi @gcstr! I'm so sorry you faced this bump on the road.

Code-signing is quite tricky in OSS. I'll try to explain the setup we have here:

* For production builds, the app is code-signed using my personal Apple Developer account which I bought to close #13. I can't share that signing certificate here because it is tied to my personal credit card. I dislike that situation, and wish I could have an organization certificate that I could share with contributors, but these are always paid-for, so linked to someone's credit card, unfortunately. So for now, my cert is signing prod build and is hidden in CI.

* For local builds, I would like people to be able to code-sign using a local self-signed certificate. In the XCode configuration, I named it `Local Self-Signed`. I automated all the steps to generate a self-signed code-signing cert that works in `ci/generate_debug_certificate.sh`. Please run it, and you will have a code-signing cert named `Local Self-Signed` in your keychain, and XCode will be able to find it and code-sign using it.

So basically, could you try the following steps and let me know if that works for you?

* Reverting the code signing settings in XCode (you changed it to `Development`). It should say `Local Self-Signed`
* Running `ci/generate_debug_certificate.sh`

It should work at that point. I'm sorry you got this annoying code-sign issue. Let me know if you're still stuck. Hopefully not! :)

**Comment by gcstr** [comment]
Hey @lwouis!

Thanks a lot for the fast and so descriptive response. It's long due for me to renew my enrolment with Apple, didn't want to expend €99 but I think it will be inevitable, wasn't sure if purchasing it would automatically help me with those permission issues, but apparently yes :)

In the meanwhile, I will try to use the self-signed certificate.
I couldn't find the file `ci/generate_debug_certificate.sh` file in the repository, is it checked in?

**Comment by lwouis** [comment]
> Thanks a lot for the fast and so descriptive response. It's long due for me to renew my enrolment with Apple, didn't want to expend €99 but I think it will be inevitable, wasn't sure if purchasing it would automatically help me with those permission issues, but apparently yes :)

Oh! I think I got you confused! You don't need to pay for a developer account to work on this project!

* For prod builds, I use my personal account certificate to code-sign. This makes the build pass Gatekeeper for users
* For local builds/runs, I introduced a self-signed certificate so that permissions persist between builds (i.e. you won't have to check/uncheck the box every build)

I think what got you confused is that I'm referring to things that don't exist on `master`! I'm sorry I've been working on the v3 branch called `v3-poc` for many weeks now and I forgot the state of `master`. V3 is almost ready; we are waiting on #38 to release it. It has **many** big changes.

Could you work on that branch instead of master? The `ci/generate_debug_certificate.sh` was added on that branch, so it only exists there where the whole setup is a bit different. Also I documented how to do all this in [the README.md from that branch](https://github.com/lwouis/alt-tab-macos/blob/v3-poc/README.md).

**Comment by gcstr** [comment]
> You don't need to pay for a developer account to work on this project!

I got it, I was just mentioning that having an Apple issued certificate would also help, It is clear that the self-signed certificate would also work.

Thanks a lot for your help again!
I'll try the script for the debug certificate now.

**Comment by gcstr** [comment]
Worked like a charm!
Thanks!
